daemon/api.go

```diff
@@ -789,26 +790,27 @@ func withEnsureUbuntuCore(st *state.State, targetSnap string, userID int, instal
 }
 
 var errModeConflict = errors.New("cannot use devmode and jailmode flags together")
+var errClassicDevmodeConflict = errors.New("cannot use classic and devmode flags together")
 var errNoJailMode = errors.New("this system cannot honour the jailmode flag")
 
-func modeFlags(devMode, jailMode bool) (snapstate.Flags, error) {
-	devModeOS := release.ReleaseInfo.ForceDevMode()
+func modeFlags(devMode, jailMode, classic bool) (snapstate.Flags, error) {
 	flags := snapstate.Flags{}
-	if jailMode {
-		if devModeOS {
-			return flags, errNoJailMode
-		}
-		if devMode {
-			return flags, errModeConflict
-		}
-		flags.JailMode = true
-	}
-	if devMode || devModeOS {
-		flags.DevMode = true
-	}
-
+	devModeOS := release.ReleaseInfo.ForceDevMode()
+	switch {
+	case jailMode && devModeOS:
+		return flags, errNoJailMode
+	case jailMode && devMode:
+		return flags, errModeConflict
```

Comment:
could you rename `errModeConflict`? So it works with `errClassicDevmodeConflict`. Or unify both into `errModeConflict` making it smarter than just a string, but that's probably overdoing it.